pkg/controller/jobframework/webhook/defaulter_test.go

```diff
@@ -87,13 +151,69 @@ func TestLossLessDefaulter(t *testing.T) {
 		t.Errorf("Response not allowed")
 	}
 	wantPatches := []jsonpatch.Operation{
-		{
-			Operation: "add",
-			Path:      "/foo",
-			Value:     "bar",
-		},
+		{Operation: "add", Path: "/creationTimestamp"},
+		{Operation: "add", Path: "/foo", Value: "foo"},
+		{Operation: "remove", Path: "/bar"},
+		{Operation: "remove", Path: "/baz"},
+		{Operation: "replace", Path: "/finalizers/0", Value: "bar"},
+		{Operation: "remove", Path: "/finalizers/1"},
+		{Operation: "remove", Path: "/labels/foo"},
+		{Operation: "replace", Path: "/subresource/foo", Value: ""},
+		{Operation: "replace", Path: "/subresource/bar"},
+		{Operation: "remove", Path: "/conditions/0/observedGeneration"},
+		{Operation: "remove", Path: "/conditions/1"},
 	}
-	if diff := cmp.Diff(wantPatches, resp.Patches); diff != "" {
+	if diff := cmp.Diff(wantPatches, resp.Patches, cmpopts.SortSlices(func(a, b jsonpatch.Operation) bool {
+		return a.Path < b.Path
+	})); diff != "" {
 		t.Errorf("Unexpected patches (-want, +got): %s", diff)
 	}
 }
+
+func TestConvertToFieldErrorPath(t *testing.T) {
```

Comment:
IMO, it's better to have more coverage in the test cases of the higher level (exported) functions.

That allows us to change the implementation (or even get rid of the internal functions) without having to change the tests.

Reply:
+1

Reply:
Done